elfi/methods/results.py

```diff
@@ -192,6 +192,21 @@ def sample_means_array(self):
         """
         return np.array(list(self.sample_means.values()))
 
+    def save_samples(self, name=None, type_='csv'):
+        """Save samples in csv or json file formats."""
+        import pandas as pd
+
+        if not name:
+            name = self.__class__.__name__.lower() + '_' + 'samples.' + type_
+
+        df = pd.DataFrame(self.samples, columns=self.samples.keys())
```

Comment:
Hmm... Seems a bit overkill to me to import pandas and convert the numpy arrays into a dataframe just for this...? How about using `np.savetxt` with appropriate formatting (could be user-givable) and header for csv.

For json there's the built-in `json` package.

Reply:
Ok, I have rewritten implementation related to `csv`.

Situation with `json` is quite different since serialization of numpy arrays in ordered dict is somehow tricky. The only solution which comes to mind is to convert numpy arrays into list and create new dict with this data. Is it okay? Of course it needs to use for loop, but since there are not too much true parameters, it will process quickly.

Reply:
I suspect (haven't checked) that pandas is doing the same internally.